src/main/java/org/springframework/data/elasticsearch/NoSuchIndexException.java

```diff
@@ -15,13 +15,11 @@
  */
 package org.springframework.data.elasticsearch;
 
-import org.springframework.dao.NonTransientDataAccessResourceException;
-
 /**
  * @author Christoph Strobl
  * @since 3.2
  */
-public class NoSuchIndexException extends NonTransientDataAccessResourceException {
+public class NoSuchIndexException extends SpringDataElasticsearchException {
```

Comment:
That exception was already fine. No need to turn this one into an uncategorized exception.